app/models/labware_creators/pcr_cycles_binned_plate.rb

```diff
@@ -62,76 +68,277 @@ def parent_v1
+    #
+    # The dilutions calculator works out how the samples will we rearranged when
+    # transferred into the child plate
+    #
+    def dilutions_calculator
+      @dilutions_calculator ||= Utility::PcrCyclesBinningCalculator.new(well_details)
+    end
+
+    def labware_wells
+      parent.wells
+    end
+
+    private
 
-      # update fields on each well with various metadata
-      fields_to_update = %w[diluent_volume pcr_cycles submit_for_sequencing sub_pool coverage]
+    #
+    # Well filter handles the selection of wells for transfer based on submission id
+    #
+    def well_filter
+      @well_filter ||= WellFilterBySubmission.new(creator: self, submission_id: submission.id)
+    end
 
-      child_wells_by_location = child_v2.wells.index_by(&:location)
+    #
+    # Validation to check that at least some wells should be transferred
+    # to catch situation where customer file has no valid rows (e.g. all zero sample volume)
+    #
+    def some_wells_are_being_transferred?
+      return true unless well_details.size.zero?
 
-      well_details.each do |parent_location, details|
-        child_position = transfer_hash[parent_location]['dest_locn']
-        child_well = child_wells_by_location[child_position]
+      errors.add(:csv_file, format(NO_WELLS_TO_TRANSFER))
+      false
+    end
 
-        update_well_with_metadata(child_well, details, fields_to_update)
-      end
+    def check_for_well_missing_detail(well)
+      return if well_details.include? well.location
+      errors.add(:csv_file, format(MISSING_WELL_DETAIL, well.location))
     end
 
-    def update_well_with_metadata(well, metadata, fields_to_update)
-      options = fields_to_update.index_with { |field| metadata[field] }
-      well.update(options)
+    def check_for_well_pending(well)
+      return unless well.pending?
+      errors.add(:csv_file, format(PENDING_WELL, well.location))
     end
 
+    #
+    # Validation to check wells have the required information and are in the right state
+    #
     def wells_have_required_information?
-      filtered_wells.each do |well|
-        next if well.aliquots.empty?
+      labware_wells.each do |well|
+        # skip validation on wells empty or chosen not to go forward (those where user has set zero sample vol)
+        next if well.aliquots.empty? || skipped_wells.include?(well.location)
 
-        errors.add(:csv_file, format(MISSING_WELL_DETAIL, well.location)) unless well_details.include? well.location
-        errors.add(:csv_file, format(PENDING_WELL, well.location)) if well.pending?
+        check_for_well_missing_detail(well)
+        check_for_well_pending(well)
       end
     end
 
-    def dilutions_calculator
-      @dilutions_calculator ||= Utility::PcrCyclesBinningCalculator.new(well_details)
+    #
+    # Creates and builds the submission, creates the child plate labware and
+    # transfers samples from the parent plate to the child plate
+    #
+    def create_labware!
```

Comment:
Possibly but I had to change this one to extend Base instead of Stamped plate because it's behaviour is now significantly different (not all wells are transferred now).